src/integrator.jl

```diff
@@ -13,7 +13,6 @@ Base.show(io::IO, l::Leapfrog) = print(io, "Leapfrog(ϵ=$(round(l.ϵ; sigdigits=
 
 Leapfrog(ϵ::T) where {T<:AbstractFloat} = Leapfrog(ϵ, zero(T))
 
-# TODO: double check the function below to see if it is type stable or not
 function step(
```

Comment:
@yebai Regarding the leapfrog step implementation, we discussed before that the current one is less optimized. It seems that combing L30 and L34 can give nontrivial improvement. I checked it via the following example.

```julia
julia> using BenchmarkTools

julia> x = rand(100);

julia> y = rand(100);

julia> eps = 0.01;

julia> f1() = x - eps * y
f1 (generic function with 1 method)

julia> f2() = (z = x - eps / 2 * y; z = z - eps / 2 * y)
f2 (generic function with 1 method)

julia> @benchmark f1()
BenchmarkTools.Trial: 
  memory estimate:  1.75 KiB
  allocs estimate:  2
  --------------
  minimum time:     140.348 ns (0.00% GC)
  median time:      189.408 ns (0.00% GC)
  mean time:        223.946 ns (8.04% GC)
  maximum time:     70.844 μs (99.36% GC)
  --------------
  samples:          10000
  evals/sample:     451

julia> @benchmark f2()
BenchmarkTools.Trial: 
  memory estimate:  3.53 KiB
  allocs estimate:  6
  --------------
  minimum time:     290.158 ns (0.00% GC)
  median time:      325.403 ns (0.00% GC)
  mean time:        400.899 ns (9.83% GC)
  maximum time:     156.173 μs (99.57% GC)
  --------------
  samples:          10000
  evals/sample:     196
```

Reply:
It is hard to tell whether this is a significant improvement without comparison to the total sampling time: as far as I know, The time for `leapfrog` might be a tiny fraction of total computation time, while most computation is spent on computing the `logpdf` and its gradients. I suggest that we keep `leapfrog` as-is` for now and change it only when more concrete benchmarking is performed.